### ISSUE TITLE
Inventory from crashed space lander not drop

### ISSUE THREAD

**Issue by FrostQ** [micdoodle8/Galacticraft#3229]
Minecraft version: 1.12.2
Galacticraft version: 4.0.1.150

Single player (SSP), Multiplayer (SMP), or SSP opened to LAN (LAN)?
Both

Galacticraft add-ons and other mods installed?
Yes

Including rocket and launch pad

Screenshot:
![2018-03-05_20 55 13](https://user-images.githubusercontent.com/36822613/36996833-0dfda3ca-20b8-11e8-8d7f-982bc5247880.png)

**Comment by Romsik788** [comment]
it seems to me that it should be so

**Comment by MJRLegends** [comment]
Yeah this is intentional you need to slow it down so it doesnt crash into the surfacr

**Comment by radfast** [comment]
Yes, sorry - it's intended!

I don't remember if Galacticraft was **always** this way, but certainly it's how it is recently.

Players who become stranded on the Moon because of this, can use `/gchouston` command to return home.

If a lot of people think it's a problem, I guess I could switch this so there is only the item loss in **Hard Mode** ... what do you guys think?

**Comment by TheQuadShot** [comment]
Mission Planning should cover any and all conceivable problems/issues that may occur... I do recall it being this way in 1.7.10 and there are vids of crashes on youtube to confirm.

**Comment by MJRLegends** [comment]
If im honest my opinion is if your not going to slow it down then yeah you should lose it on crashing. and theres the new command if they need to get home.

**Comment by MJRLegends** [comment]
or maybe you could make it so it a random chance something will be saved either some of the pads or rocket or some of the cargo from the rocket but will be completely random.

This will give the element of something surviving the crash but not all. or you could make it so you get a scrap material back which can be turned in to actual resources, like to give players some items like compressed steel back to remake things

**Comment by TheQuadShot** [comment]
Currently your inventory is on the ground, just like on the overworld and you get blown-up by a creeper. I'm not sure, but if you did have storage on the rocket, then that inventory should drop just like a chest (blown up by creeper) on the overworld. The biggest issue I "see" is that you spawn at 0, 0 and the ship/crash site isn't at 0, 0. Not to mention that it's a mistake that usually happens only once. On multi-player getting your oxygen gear and getting rescued is a pretty easy process. I would add that if they did remember to pack an **Space Emergency Kit** it should survive the crash and have a long despawn rate.

**Comment by radfast** [comment]
Thanks for the further suggestions.

The problem is that getting home from the Moon is dependent on the player having 6 specific things:
- rocket
- 9 x rocket launch pad
- fuel
- Fuel Loader
- power for the Fuel Loader
- a container to put the fuel in (Canister is best)

The first three things in the list are contained in the Lander.   If the Lander crashes then I think it's only right that some of these things are lost - for example, the fuel would be lost.  I don't see the advantage of giving the player back a random selection of the items.  For example, give the player 6 x rocket launch pad and he is still stranded just as much as before (unless maybe he brought 3 spare rocket launch pads?)

The final three things in the list, the player is supposed to figure out that he needs to carry them with him in player inventory, or - if not - he can get them on the Moon from mining sapphires and then trading with Moon Villagers.  (If he needs to buy multiple items from the Moon Villagers, that's going to be a **lot** of mining so he can probably burn through 2 pickaxes at least and several oxygen tanks and food to do all that mining....)

TL;DR: I think it actually makes Galacticraft **more fun** if you have a real risk of player death and item loss when you do something dumb.  Crash landing on the Moon is dumb.  Skilled players do not have many other challenges in Minecraft - for example every mob is easy to defeat once you have good armor and weapons.  First landing on the Moon should feel dangerous!

**Comment by EzerArch** [comment]
> Players who become stranded on the Moon because of this, can use /gchouston command to return home.

With:

> 3. (...) i.e. someone who does not read the guidance on the Wiki etc - is going to know about that command. I guess the game could **detect a looped dying pattern** and prompt for that command in chat. (Don't want to automate it totally, because the looped dying could be for another reason, e.g. a player attack on a person's Moon home, on a griefing/raiding server). (from: https://github.com/micdoodle8/Galacticraft/issues/3184#issuecomment-368111154)

If I may, I would add two or tree criteria to the triggering pattern:
1. Lander crash triggers an internal 2-minute countdown;
2. Player died 2 times within the countdown;
3. No items in the player's inventory (a sign the player has given up trying).

**Comment by TheQuadShot** [comment]
I think the biggest problem is the respawn, sometimes the player is lost or just too far away from the crash site to get there in time. Although I don't know how you could do this... It would be nice if: the player did have a Space Emergency Kit in their inventory, have them respawn with it in their inventory. The biggest problem I can think of is, if they have a grave type mod.

**Comment by radfast** [comment]
>  If I may, I would add two or tree criteria to the triggering pattern:

    Lander crash triggers an internal 2-minute countdown;
    Player died 2 times within the countdown;
    No items in the player's inventory (a sign the player has given up trying).

Nice :)

It's kind of roughly what the game is doing currently.  The game looks for 2 deaths within 1 minute 15 seconds, on any planet, then gives a chat message, "Umm, Houston, we've had a problem".  1 minute 15 seconds is approximately the survival time without oxygen in Easy mode, exact time depends on configs and hunger situation, if you're jumping or running it will be less.  If you think we need to change that time please make suggestions...

No items is a tricky test, because in a lander crash there's a lot of loose blocks lying around so a player can very easily pick up some blocks.

>   It would be nice if: the player did have a Space Emergency Kit in their inventory, have them respawn with it in their inventory. The biggest problem I can think of is, if they have a grave type mod.

OK this makes a lot of sense.  I don't play with Gravestone myself but I can see why this would be appreciated by the players who do.  The Emergency Kit is designed to let you get back to the death location so that you can retrieve items - even if you don't have a Gravestone mod (I guess really the Gravestone mod is only saving you on that 5 minute dropped items timer - handy on servers which also run auto-deletes on dropped items sometimes.)

Also wondering if we should change the Potion of Healing to a Potion of Speed?  Healing's not really necessary when you respawn on full health, food and a weapon are more important.
